src/pcp/meminfo/pcp-meminfo.py

```diff
@@ -222,9 +223,11 @@ def __init__(self):
         mngr["meminfo"] = METRICS
         mngr["sysinfo"] = SYS_METRICS
         mngr.printer = MeminfoReport(opts)
-        sts = mngr.run()
-        sys.exit(sts)
-
+        try:
+            sts = mngr.run()
+            sys.exit(sts)
+        except IOError:
+            signal.signal(signal.SIGPIPE,signal.SIG_DFL)
```

Comment:
@orasagar hmm, I don't understand what this does.

Do you know where the SIGPIPE is coming from?  (I guess its related to the connection to pmcd?  but libpcp should be ensuring that doesn't propogate up to this level).  Is it reproducible somehow?

Finally, isn't setting the handler to sig_default inside the exception handler is too late?  Or is this a nested failure?